components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -1186,7 +1186,7 @@ private IdentityProviderListResponse createIDPListResponse(IdpSearchResult idpSe
         }
 
         listResponse.setTotalResults(idpSearchResult.getTotalIDPCount());
-        listResponse.setStartIndex(idpSearchResult.getOffSet() + 1);
+        listResponse.setStartIndex(idpSearchResult.getOffSet());
```

Comment:
the start index of the search result should be always offset + 1. Why was this changed?

Reply:
According to the issue description start index should be the provided offset

Reply:
startIndex of the pagination result should identify the index of the first element in a page. Therefore it should be offset + 1 always. E.g. refer https://github.com/wso2/identity-api-user/blob/master/components/org.wso2.carbon.identity.api.user.application/org.wso2.carbon.identity.rest.api.user.application.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/application/v1/core/ApplicationService.java#L131